appservice/src/tree/DeploymentsTreeItem.ts

```diff
@@ -25,15 +32,17 @@ export class DeploymentsTreeItem extends AzExtParentTreeItem {
     public readonly childTypeLabel: string = localize('Deployment', 'Deployment');
     public readonly site: ParsedSite;
     public suppressMaskLabel: boolean = true;
+    public readonly contextValuesToAdd: string[];
 
     private _scmType?: string;
     private _repoUrl?: string;
 
-    public constructor(parent: AzExtParentTreeItem, site: ParsedSite, siteConfig: SiteConfig, sourceControl: SiteSourceControl) {
+    public constructor(parent: AzExtParentTreeItem, options: DeploymentsTreeItemOptions) {
         super(parent);
-        this.site = site;
-        this._scmType = siteConfig.scmType;
-        this._repoUrl = sourceControl.repoUrl;
+        this.site = options.site;
+        this._scmType = options.siteConfig.scmType;
+        this._repoUrl = options.sourceControl.repoUrl;
+        this.contextValuesToAdd = options?.contextValuesToAdd ?? [];
```

Comment:
I think this is a misuse of the nullish coalescing operator. We don't want any falsy values other than `[]` making their way into `contextValuesToAdd`

Reply:
If someone passed in a falsy value here and ignored the typings, I think I'd want it to throw an error (which it would when we do `...this.contextValuesToAdd`.

If they obey the typings, the left side of that expression can only be `string[] | undefined` so there will never be a falsy value.

Reply:
> If someone passed in a falsy value here and ignored the typings, I think I'd want it to throw an error (which it would when we do ...this.contextValuesToAdd.

I don't think throwing an error in this case would be the best course of action.

If a falsy value (other than `[]`) was somehow passed in, it was definitely a mistake and not worth erroring. Especially because it would likely be an ugly error like "found non-callable iterator". The `??` operator is great for some situations but `||` is a better fit here.

I bring this up not only for this situation but as something to keep in mind for future code too. In my estimation the number of times we'll need to use `??` is relatively small compared to the need for `||`.